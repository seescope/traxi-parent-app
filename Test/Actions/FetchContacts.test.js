import fetchContacts from '../../App/Actions/FetchContacts';

describe('fetchContacts', () => {
  it('loads the contacts and dispatches a GOT_CONTACTS event', () => {
    const TEST_DISPATCH = jest.fn();

    // Exercise
    const action = fetchContacts();
    const promise = action(TEST_DISPATCH);

    return promise.then(() => {
      expect(TEST_DISPATCH.mock.calls).toMatchSnapshot();
    });
  });
});

