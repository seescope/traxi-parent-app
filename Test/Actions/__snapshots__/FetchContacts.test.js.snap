exports[`fetchContacts loads the contacts and dispatches a GOT_CONTACTS event 1`] = `
Array [
  Array [
    Object {
      "contacts": Array [
        Object {
          "avatarURL": "content://com.android.contacts/contacts/1005/photo",
          "name": "Quak Yin",
          "phoneNumber": "+60142837979",
        },
        Object {
          "avatarURL": "content://com.android.contacts/contacts/1005/photo",
          "name": "Marc Robinson",
          "phoneNumber": "+61400669057",
        },
        Object {
          "avatarURL": "content://com.android.contacts/contacts/688/photo",
          "name": "Guy Lupo",
          "phoneNumber": "+61432031301",
        },
      ],
      "type": "GOT_CONTACTS",
    },
  ],
]
`;
